### PR TITLE
Add Spanish translation for upgrading.md

### PR DESCRIPTION
**Pull request by alemohamad**
Add Spanish translation for upgrading.md

### CODE REVIEW

File: docs/upgrading.es.md [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
+```diff
+- public func routes(_ router: Router, _ container: Container) throws
++ public func routes(_ app: Application) throws
+```
+
+A continuación se menciona más información sobre los cambios en la sintaxis de routing.
+
+## Servicios
+
+Vapor 4's services APIs have been simplified to make it easier for you to discover and use services. Services are now exposed as methods and properties on `Application` and `Request` which allows the compiler to help you use them.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:182]:
```suggestion
```

File: docs/upgrading.es.md [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
+- public func routes(_ router: Router, _ container: Container) throws
++ public func routes(_ app: Application) throws
+```
+
+A continuación se menciona más información sobre los cambios en la sintaxis de routing.
+
+## Servicios
+
+Vapor 4's services APIs have been simplified to make it easier for you to discover and use services. Services are now exposed as methods and properties on `Application` and `Request` which allows the compiler to help you use them. 
+Las APIs de servicios de Vapor 4 se han simplificado para que resulte más fácil descubrir y utilizar los servicios. Los servicios ahora están expuestos como métodos y propiedades en `Application` y `Request`, lo que permite al compilador ayudar a usarlos.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:183]:
```suggestion
Las APIs de servicios de Vapor 4 se han simplificado para que resulte más fácil descubrirlos y utilizarlos. Los servicios ahora están expuestos como métodos y propiedades en `Application` y `Request`, lo que permite al compilador ayudarte en su uso.
```

File: docs/upgrading.es.md [hunk 3]
```diff
@@ -0,0 +1,815 @@
...
++ app.leaf.cache.isEnabled = false
+```
+
+### Entorno
+
+Se puede acceder al entorno actual (producción, desarrollo, etc.) a través de `app.environment`.
+
+### Servicios Personalizados
+
+Custom services conforming to the `Service` protocol and registered to the container in Vapor 3 can be now be expressed as extensions to either Application or Request.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:251]:
```suggestion
```

File: docs/upgrading.es.md [hunk 4]
```diff
@@ -0,0 +1,815 @@
...
+print(app.myNumber) // 42
+```
+
+## NIO
+
+Vapor 4 ahora expone las APIs asíncronas de SwiftNIO directamente y no intenta sobrecargar métodos como `map` y` flatMap` o tipos de alias como `EventLoopFuture`. Vapor 3 proporcionó sobrecargas y alias para compatibilidad con versiones beta anteriores que se lanzaron antes de que existiera SwiftNIO. Estos se han eliminado para reducir la confusión con otros paquetes compatibles con SwiftNIO y seguir las recomendaciones de mejores prácticas de SwiftNIO.
+
+### Cambios de nombres asíncronos
+
+El cambio más obvio es que se ha eliminado el alias de tipo `Future` en pos de `EventLoopFuture`. Esto se puede solucionar con bastante facilidad buscando y reemplazando.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:323]:
```suggestion
El cambio más obvio es la eliminación del alias de tipo `Future` para `EventLoopFuture`. Esto se puede solucionar con bastante facilidad buscando y reemplazando.
```

File: docs/upgrading.es.md [hunk 5]
```diff
@@ -0,0 +1,815 @@
...
+
+Para convertir un `ByteBuffer` nuevamente en `Data`, usa:
+
+```swift
+Data(buffer.readableBytesView)
+```
+
+### Throwing map / flatMap
+
+El cambio más difícil es que `map` y `flatMap` ya no se pueden devolver en throw. `map` tiene una versión de throw llamada (de manera algo confusa) `flatMapThrowing`. Sin embargo, `flatMap` no tiene contraparte de throw. Esto puede requerir que tengas que reestructurar algo de código asincrónico.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:362]:
```suggestion
El cambio más difícil es que `map` y `flatMap` ya no lanzan errores (throw). `map` tiene una versión throw llamada (de manera algo confusa) `flatMapThrowing`. Sin embargo, `flatMap` no tiene contrapartida throw. Esto puede requerir que tengas que reestructurar algo de código asincrónico.
```

File: docs/upgrading.es.md [hunk 6]
```diff
@@ -0,0 +1,815 @@
...
+```
+
+El middleware ya no se puede registrar por nombre de tipo. Primero debes inicializar el middleware antes de registrarlo.
+
+```diff
+- middleware.use(ErrorMiddleware.self)
++ app.middleware.use(ErrorMiddleware.default(environment: app.environment))
+```
+
+Para eliminar todo el middleware predeterminado, configura `app.middleware` en una configuración vacía usando:
```
Comment (jacostaf10) [inline docs/upgrading.es.md:498]:
```suggestion
Para eliminar todo el middleware predeterminado, establece `app.middleware` en una configuración vacía usando:
```

File: docs/upgrading.es.md [hunk 7]
```diff
@@ -0,0 +1,815 @@
...
+
+Todos los modelos deben tener su tabla o nombre de colección definido estáticamente.
+
+```diff
+final class Planet: Model {
++   static let schema = "Planet"    
+}
+```
+
+Todos los modelos ahora deben tener un inicializador vacío. Dado que todas las propiedades usan property wrappers, esto puede estar vacío.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:549]:
```suggestion
Todos los modelos ahora deben tener un inicializador vacío. Dado que todas las propiedades usan property wrappers, puede estar vacío.
```

File: docs/upgrading.es.md [hunk 8]
```diff
@@ -0,0 +1,815 @@
...
+
+```diff
+final class Planet: Model {
++   init() { }
+}
+```
+
+`save`, `update`, y `create` del modelo ya no devuelven la instancia del modelo.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:557]:
```suggestion
Los métodos `save`, `update`, y `create` del modelo ya no devuelven la instancia del modelo.
```

File: docs/upgrading.es.md [hunk 9]
```diff
@@ -0,0 +1,815 @@
...
+try app.databases.use(.postgres(url: "postgres://..."), as: .psql)
+```
+
+`MigrationsConfig` ha sido reemplazado por `app.migrations`.
+
+```swift
+app.migrations.use(CreatePlanet(), on: .psql)
+```
+
+### Repositories
```
Comment (jacostaf10) [inline docs/upgrading.es.md:710]:
```suggestion
### Repositorios
```

File: docs/upgrading.es.md [hunk 10]
```diff
@@ -0,0 +1,815 @@
...
+
+Como la forma en que funcionan los servicios en Vapor 4 ha cambiado, eso también significa que la forma de hacer repositorios de bases de datos ha cambiado. Aún necesitas un protocolo como `UserRepository`, pero en lugar de hacer que una `final class` se ajuste a ese protocolo, debes crear una `struct`.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:712]:
```suggestion
Como la forma en que funcionan los servicios en Vapor 4 ha cambiado, la manera de hacer repositorios de bases de datos también. Aún necesitas un protocolo como `UserRepository`, pero en lugar de hacer que una `final class` se ajuste a ese protocolo, debes crear una `struct`.
```

File: docs/upgrading.es.md [hunk 11]
```diff
@@ -0,0 +1,815 @@
...
+- final class DatabaseUserRepository: UserRepository {
++ struct DatabaseUserRepository: UserRepository {
+      let database: Database
+      func all() -> EventLoopFuture<[User]> {
+          return User.query(on: database).all()
+      }
+  }
+```
+
+También debes eliminar la conformidad de `ServiceType`, ya que ya no existe en Vapor 4.
```
Comment (jacostaf10) [inline docs/upgrading.es.md:724]:
```suggestion
También debes eliminar la conformidad de `ServiceType`, puesto que ya no existe en Vapor 4.
```

**Review by jacostaf10 — CHANGES_REQUESTED** [review]
Deleted some text  left behind and made a few changes to improve readability

**Comment by alemohamad** [comment]
Thanks @jacostaf10 !!!

**Review by jacostaf10 — APPROVED** [review]
All right now.

**Pull request merged**